TinCan/InteractionType.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace TinCan
+{
+    public sealed class InteractionType
+    {
+        private const string choice = "choice";
+        private const string sequencing = "sequencing";
+        private const string likert = "likert";
+        private const string matching = "matching";
+        private const string performance = "performance";
+        private const string truefalse = "true-false";
+        private const string fillin = "fill-in";
+        private const string numeric = "numeric";
+        private const string other = "other";
+
+
+        public static readonly InteractionType Choice = new InteractionType("choice");
+        public static readonly InteractionType Sequencing = new InteractionType("sequencing");
+        public static readonly InteractionType Likert = new InteractionType("likert");
+        public static readonly InteractionType Matching = new InteractionType("matching");
+        public static readonly InteractionType Performance = new InteractionType("performance");
+        public static readonly InteractionType TrueFalse = new InteractionType("true-false");
+        public static readonly InteractionType FillIn = new InteractionType("fill-in");
+        public static readonly InteractionType Numeric = new InteractionType("numeric");
+        public static readonly InteractionType Other = new InteractionType("other");
+
+
+        private InteractionType(string value)
+        {
+            Value = value;
+        }
+
+        public static InteractionType FromValue(string value)
+        {
+            switch (value)
+            {
+                case choice:
```

Comment:
Can cases use a constant?

Reply:
They can, because a const's value is known at compile time.
(This is in contrast to readonly variables, they are not initialised until runtime and thus not known at compile time.)